<launch>

<!-- Enable/disable stuff-->
  <arg name="run_peopledetector"  default="false"/>
  <arg name="run_compressor_remote"      default="false"/>
  <arg name="run_compressor_local"      default="true"/>

<!--  Compressor parameters-->

  <arg name="voxel_size" default="0.04"/>
  <arg name="crop"      default="true"/>
  <arg name="min_x"     default="0.5"/>
  <arg name="max_x"     default="9.5"/>
  <arg name="min_y"     default="0.5"/>
  <arg name="max_y"     default="9.5"/>
  <!--<arg name="min_z"     default="0.1"/>-->
  <arg name="min_z"     default="-0.1"/>
  <arg name="max_z"     default=" 4.3"/>

<!--  ICRA2020 parameters Compressor, AAA -->
<!--<arg name="run_compressor"    default="true"/>
<arg name="voxel_size" default="0.04"/>
<arg name="crop"      default="true"/>
<arg name="min_x"     default="1.5"/>
<arg name="max_x"     default="7.5"/>
<arg name="min_y"     default="3.3"/>
<arg name="max_y"     default="9.0"/>
<arg name="min_z"     default="0.5"/>
<arg name="max_z"     default=" 3.0"/>-->

<!--  Decompressor-->
<!--  <arg name="filter_value"        default="127" /> -->
  <arg name="filter_value"        default="80" /> <!--decompressor [0-255]-->


<machine name="master" address="localhost"/>

<!-- ================= Compressor (remote) ======================= -->
<group if="$(arg run_compressor_remote)">
  <!-- stuff that will only be evaluated if foo is true -->
<!--    #define _MINX 0.0
    #define _MINY 0.0
    #define _MINZ 0.1
    #define _MAXX 10.0
    #define _MAXY 10.0
    #define _MAXZ 4.0-->

<node pkg="wp3_compressor" type="wp3_compressor_node" name="jetson_comp_kinect" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/jetson/wp3/points_nocolor_merged" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/jetson/wp3/points_merged_comp" />
    <param name="local_frame" value="world" />
<!--    <param name="global_frame" value="jetson1_ir_optical_frame" />-->
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

</group>



<!-- ================= Compressor (local) ======================= -->
<group if="$(arg run_compressor_local)">

<node pkg="wp3_compressor" type="wp3_compressor_node" name="jetson_comp_kinect" machine="master">
    <param name="resolution" value="$(arg voxel_size)" />
    <param name="input_topic" value="/jetson/wp3/points_nocolor_merged" />
    <param name="input_type" value="0" />
    <param name="output_topic" value="/jetson/wp3/kinect_comp" />
    <param name="local_frame" value="world" />
    <param name="global_frame" value="world" />
    <param name="crop"      value="$(arg crop)" />
    <param name="min_x"     value="$(arg min_x)" />
    <param name="max_x"     value="$(arg max_x)" />
    <param name="min_y"     value="$(arg min_y)" />
    <param name="max_y"     value="$(arg max_y)" />
    <param name="min_z"     value="$(arg min_z)" />
    <param name="max_z"     value="$(arg max_z)" />
</node>

</group>

<node pkg="wp3_decompressor" type="wp3_decompressor_node" name="jetson_decomp_kinect">
<!--    <param name="input_topic" value="/jetson/wp3/points_merged_comp" />-->
<!--    <param name="output_topic" value="/jetson/wp3/points_merged" />-->
    <param name="sensor_name" value="jetson" />
    <param name="sensor_frame" value="world" />
    <param name="sensor_type" value="0" />
    <param name="filter_value" value="$(arg filter_value)" />
</node>

<!-- ================= People Detector (remote) ======================= -->
<group if="$(arg run_peopledetector)">
  <include file="$(find darknet_ros)/launch/all_nodes.launch"/>
</group>


</launch>
